lib/std/os.zig

```diff
@@ -6869,6 +6869,7 @@ pub fn tcgetpgrp(handle: fd_t) TermioGetPgrpError!pid_t {
             .SUCCESS => return pgrp,
             .BADF => unreachable,
             .INVAL => unreachable,
+            .INTR => continue,
```

Comment:
man tcgetpgrp does not show EINTR and neither do other posix implementation man pages. It sounds abit weird, that querying of a simple process value would be non-atomic for the duration of the call as the underlying information is `pid_t = i32`.

Reply:
I had the exact same thought, "how did *this* get interrupted". But given I did get it back as an `unexpectedErrno` I thought it was best to add it back.

Reply:
> I can try to put together a repro/test if you don't believe me?

Ideally yes, but at least describing the exact context sounds helpful. Since this is undocumented and sounds awkward it should be also a brief comment.